### DISCUSSION TITLE
x-trap noscroll shifts modal left when closing

### DISCUSSION THREAD

**Discussion by thecodeclimber** [alpinejs/alpine#2690] [7. Bugs (archived)] [answered]
It seems that `x-trap.noscroll` causes the  panel of a modal to shift left when closing.

I have tested this out using the code from the Modal component on this page:  https://alpinejs.dev/component/modal

The only change I made was I used `w-1/2` instead of `w-full`.  If you try that out and close the modal you will see the panel pane shift when closing.  If you remove `noscroll` from `x-trap` then you will notice that it does NOT shift when closing.

Any idea how to fix this?

Thanks :).

**Comment by magnusskare** [comment, marked as answer]
This is not an issue with Alpine, but basically how `overflow: hidden` works. `noscroll` sets `overflow: hidden` on the `HTML`-tag, which removes scrollbars if you have them set to be visible. This causes elements to shift relative to the scrollbar width, but is less noticeable on devices with hidden scrollbars (MacOS by default, iOS etc.).

**Reply by thecodeclimber** [reply]
Thanks - that is very helpful.

Do you know of any way in css to deal with that issue?

**Reply by magnusskare** [reply]
There's quite a few ways to handle this, but my guess is that this behavior is familiar to users with active scrollbars anyway. There's also accessibility-concerns if you want to keep the scrollbars visible, while removing functionality.

Assuming your modal is positioned `fixed` or `absolute`, you could set the `left` value on load (and resize/orientation) with JavaScript. This should prevent the modal itself from jumping (assuming the scrollbar is on the right side), but everything else relative to the viewport would naturally shift when the scrollbar is removed.
